saq/queue.py

```diff
@@ -260,16 +260,16 @@ async def sweep(self, lock=60):
                     logger.info("Sweeping job %s", job)
         return swept
 
-    async def listen(self, job_ids, callback, timeout=10):
+    async def listen(self, job_keys, callback, timeout=10):
         """
         Listen to updates on jobs.
 
-        job_ids: sequence of job ids
-        callback: callback function, if it returns truthy, break
+        job_keys: sequence of job keys
+        callback: callback function, if it returns truthy, break. Takes (job_id, status) as arguments
         timeout: if timeout is truthy, wait for timeout seconds
         """
         pubsub = self.redis.pubsub()
-
+        job_ids = [Job.id_from_key(job_key, self.name) for job_key in job_keys]
```

Comment:
i don't like having to do id_from_key, can we remove this method? can't we just do something like

queue.job_id(job_key)

because you can never call Job.id_from_key without the queue anyways, so forget the Job and call the queue

Reply:
@barakalon thoughts?

Reply:
Makes sense to me 👍